app/src/main/java/com/sun/colornotetaking/data/local/source/LabelLocalDataSource.java

```diff
@@ -0,0 +1,66 @@
+package com.sun.colornotetaking.data.local.source;
+
+import android.content.Context;
+
+import com.sun.colornotetaking.data.OnDataLoadingCallback;
+import com.sun.colornotetaking.data.local.GetDataHandler;
+import com.sun.colornotetaking.data.local.LocalGetDataAsync;
+import com.sun.colornotetaking.data.local.dao.LabelDAO;
+import com.sun.colornotetaking.data.local.dao.LabelDAOImpl;
+import com.sun.colornotetaking.data.model.Label;
+
+import java.util.List;
+
+public class LabelLocalDataSource implements LabelDataSource {
+
+    private static LabelLocalDataSource sLabelLocalDataSource;
+    private LabelDAO mLabelDAO;
+
+    private LabelLocalDataSource(LabelDAO labelDAO) {
+        mLabelDAO = labelDAO;
+    }
+
+    public static LabelLocalDataSource getInstance(Context context) {
```

Comment:
tuơng tự inject sai r
```suggestion
    public static LabelLocalDataSource getInstance(LabelDAO lableDAO) {
```